src/twisted/internet/testing.py

```diff
@@ -669,8 +669,16 @@ def callWhenRunning(
         """
         Fake L{IReactorCore.callWhenRunning}.
         Keeps a list of invocations to make in C{self.whenRunningHooks}.
+
+        If the reactor has not started, the callable will be scheduled
+        to run when it does start. Otherwise, the callable will be invoked
+        immediately.
         """
-        self.whenRunningHooks.append((callable, args, kw))
+        if self.running:
+            callable(*args, **kw)
+            return None
+        else:
+            self.whenRunningHooks.append((callable, args, kw))
```

Comment:
For reference, this matches the `ReactorBase.callWhenRunning` implementation:

https://github.com/twisted/twisted/blob/ef6160aa2595adfba0c71da6db65b7a7252f23e9/src/twisted/internet/base.py#L893-L905